graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,391 @@
+        else:
+            msg = "max_nominations must be an integer >= 1"
+            raise ValueError(msg)
+
+        # Error checking of these will be handled by GMP
+        if isinstance(graph_match_kws, dict):
+            self.graph_match_kws = graph_match_kws
+        else:
+            msg = "graph_match_kws must be type dict"
+            raise ValueError(msg)
+
+    def fit(self, A, B, voi, seeds):
+        """
+        Fits the model to two graphs.
+
+        Parameters
+        ----------
+        A: 2d-array, square
+            Adjacency matrix, the graph where ``voi`` is known
+
+        B: 2d-array, square
+            Adjacency matrix, the graph where ``voi`` is not known
+
+        voi: int
+            Vertex of interest
+
+        seeds: list
+            List of length two `[seedsA, seedsB]` where first element is
+            the seeds associated with adjacency matrix A
+            and the second element the adjacency matrix associated with B, note
+            `len(seedsA)==len(seedsB)`. The elements of `seedsA` and `seedsB` are
+            vertices which are known to be matched, that is, `seedsA[i]` is matched
+            to vertex `seedsB[i]`.
+
+        Returns
+        -------
+        self: A reference to self
+        """
+        if not isinstance(A, np.ndarray) or not isinstance(B, np.ndarray):
+            msg = "Adjacency matrices must be type np.ndarray"
+            raise ValueError(msg)
+        elif A.ndim != 2 or B.ndim != 2:
+            msg = "Adjacency matrix entries must be two-dimensional"
+            raise ValueError(msg)
+        elif A.shape[0] != A.shape[1] or B.shape[0] != B.shape[1]:
+            msg = "Adjacency matrix entries must be square"
+            raise ValueError(msg)
+
+        if not isinstance(voi, int):
+            msg = "Voi must be an integer"
+            raise ValueError(msg)
+        elif voi < 0 or voi >= A.shape[0]:
+            msg = "Voi must be in range[0, num_verts_A)"
+            raise ValueError(msg)
+
+        if not isinstance(seeds, list):
+            msg = "seeds must be a list"
+            raise ValueError(msg)
+        if len(seeds) != 2:
+            msg = "seeds must be length two, with first element containing seeds \
+                  of A and the second containing seeds of B"
+            raise ValueError(msg)
+
+        seedsA = seeds[0]
+        seedsB = seeds[1]
+
+        if not (isinstance(seedsA, list) or isinstance(seedsA, np.ndarray)) or not (
+            isinstance(seedsB, list) or isinstance(seedsB, np.ndarray)
+        ):
+            msg = "seeds[0] and seeds[1] must both be lists or arrays"
+            raise ValueError(msg)
+        if len(seedsA) != len(seedsB):
+            msg = "Must have the same number of seeds for each adjacency matrix"
+            raise ValueError(msg)
+        if len(seedsA) == 0:
+            msg = "Must include at least one seed to produce nomination list"
+            raise ValueError(msg)
+
```

Comment:
```suggestion
elif (partial_match >= len(A)).any():
        msg = "`partial_match` entries must be less than number of nodes"
    elif not len(set(partial_match[:, 0])) == len(partial_match[:, 0]) or not len(
        set(partial_match[:, 1])
    ) == len(partial_match[:, 1]):
        msg = "`partial_match` column entries must be unique"
        
   elif partial_match.shape[0] > A.shape[0]:
        msg = "`partial_match` can have only as many seeds as there are nodes"
```
 partial_match is seeds here. add these checks here and in tests